docs/libcurl/opts/CURLOPT_SSL_CERT_DIST_NAMES_FUNCTION.3

```diff
@@ -0,0 +1,52 @@
+.\" **************************************************************************
+.\" *                                  _   _ ____  _
+.\" *  Project                     ___| | | |  _ \| |
+.\" *                             / __| | | | |_) | |
+.\" *                            | (__| |_| |  _ <| |___
+.\" *                             \___|\___/|_| \_\_____|
+.\" *
+.\" * Copyright (C) 1998 - 2018, Daniel Stenberg, <daniel@haxx.se>, et al.
+.\" *
+.\" * This software is licensed as described in the file COPYING, which
+.\" * you should have received as part of this distribution. The terms
+.\" * are also available at https://curl.haxx.se/docs/copyright.html.
+.\" *
+.\" * You may opt to use, copy, modify, merge, publish, distribute and/or sell
+.\" * copies of the Software, and permit persons to whom the Software is
+.\" * furnished to do so, under the terms of the COPYING file.
+.\" *
+.\" * This software is distributed on an "AS IS" basis, WITHOUT WARRANTY OF ANY
+.\" * KIND, either express or implied.
+.\" *
+.\" **************************************************************************
+.\"
+.TH CURLINFO_TLS_CERT_DIST_NAMES_FUNCTION 3 "11 May 2021" "libcurl 7.77.0" "curl_easy_setopt options"
+
+.SH NAME
+CURLOPT_SSL_CERT_DIST_NAMES_FUNCTION \- user callback for returning distinguished names of  a client certificate request
+.SH SYNOPSIS
+.nf
+#include <curl/curl.h>
+
+CURLcode dist_names_callback(void *distNames, void *userp);
+
+CURLcode curl_easy_setopt(CURL *handle, CURLOPT_SSL_CERT_DIST_NAMES_FUNCTION,
+                         dist_names_callback);
+.SH DESCRIPTION
+Pass a pointer to your callback function, which should match the prototype
+shown above.
```

Comment:
Why is `distNames` a `void *` and what exactly does it point to when the callback gets called? You don't explain that here.

Reply:
distNames tends to be different structures in different OSes. In case of SChannel, it is SecPkgContext_IssuerListInfoEx *. https://docs.microsoft.com/en-us/windows/win32/api/schannel/ns-schannel-secpkgcontext_issuerlistinfoex

Reply:
Usage we have is:
1. Set curl options  curl_easy_setopt(m_curl, CURLOPT_SSL_DIST_NAMES_FUNCTION, DistNamesCallback)
2. When callback comes back, 
CURLcode  DistNamesCallback(void* distNames, void* state){
        if(s_backend == CURLSSLBACKEND_SCHANNEL){ 
               const auto issuerList = reinterpret_cast<const SecPkgContext_IssuerListInfoEx*>(distNames);
               ....
      }
}

Reply:
Don't you think that should be mentioned in the docs for the callback? Seems rather vital to me...

Reply:
But more importantly: that would add yet another callback that isn't TLS backend agnostic and that's **bad**. It would be better if libcurl extracted the info and provided in the callback so that the callback would work the same no matter which TLS backend that's used!

Reply:
I want that but not my knowledge on the TLS backends are very limited, that lack of knowledge hinders the will.